sympy/tensor/tensor.py

```diff
@@ -4199,3 +4201,79 @@ def _expand(expr, **kwargs):
         return expr._expand(**kwargs)
     else:
         return expr.expand(**kwargs)
+
+def Add_hook(*args, **options):
+    evaluate = options.get('evaluate', global_parameters.evaluate)
+    if not evaluate:
+        return TensAdd(*args)
+
+    extras, tensors, others = [], [], []
+    for a in args:
+        if isinstance(a, MatrixExpr):
+            extras.append(a)
+        elif isinstance(a, TensExpr):
+            tensors.append(a)
+        else:
+            others.append(a)
+    extra = Add(*extras, evaluate=True)
+    coeff = Add(*others, evaluate=True)
+
+    for i,a in enumerate(tensors):
+        if i == 0:
+            tensor = a
+        else:
+            tensor = TensAdd(tensor, a).doit()
+
+    return TensAdd(tensor, coeff, extra).doit()
+
+def Mul_hook(*args, **options):
+    evaluate = options.get('evaluate', global_parameters.evaluate)
+    if not evaluate:
+        return TensMul(*args)
+
+    for i,a in enumerate(args):
+        if i == 0:
+            result = a
+        else:
+            result = TensMul(result, a).doit()
+
+    return result
+
+def Pow_hook(b, e, **options):
+    if isinstance(e, TensExpr):
+        raise NotImplementedError
+    if e.is_negative:
+        raise ValueError('Cannot divide by a tensor')
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=SymPyDeprecationWarning)
```

Comment:
Use the `ignore_warnings` or `warns_deprecated_sympy` context managers for this.

Reply:
Got it done. Thanks!